test/explicit-length-check.js

```diff
@@ -0,0 +1,65 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/explicit-length-check';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{
+	ruleId: 'explicit-length-check',
+	message: '`length` property should be compared to a value'
+}];
+
+ruleTester.run('explicit-length-check', rule, {
+	valid: [
+		'array.foo',
+		'array.length',
+		'array.length === 0',
+		'array.length !== 0',
+		'array.length > 0',
+		'if (array.foo) {}',
+		'if (length) {}',
+		'if ([].length > 0) {}',
+		'if ("".length > 0) {}',
+		'if (array.length === 0) {}',
+		'if (array.length !== 0) {}',
+		'if (array.length !== 0 && array[0] === 1) {}'
+	],
+	invalid: [
+		{
+			code: 'if ([].length) {}',
+			errors
```

Comment:
> That seems like a documentation problem though.

Sure, but in this case, I'd rather stick to what the core RuleTester allows you to do. So that beginners and people familiar with the core RuleTester do not get confused.

This would be a lot more interesting if it allows setting any default options (environment options for instance), and then also the simple string thing I mentioned.

> Maybe try adding it to ESLint RuleTester first

Sounds good. I'm still a bit torn on this, do you feel like creating an issue?